src/gui/progressbarpainter.cpp

```diff
@@ -58,7 +58,7 @@ void ProgressBarPainter::paint(QPainter *painter, const QStyleOptionViewItem &op
     styleOption.textVisible = true;
     // QStyleOption fields
     styleOption.rect = option.rect;
-    styleOption.state = option.state;
+    styleOption.state = option.state | QStyle::State_Horizontal;
```

Comment:
Can you add a comment about this workaround for Qt6?

Reply:
I can, though I doubt it's just a '"workaround for Qt 6". `QStyle::State_Horizontal` was explicitly set for some reason before you drop/miss it out.
@jagannatharjun, have you to say something about? Is it needed for some other cases?

Reply:
if you check the commit that introduced this "bug" it seems  they require this flag https://github.com/qt/qtbase/commit/add8262f72befc4c6a25403a21269030111054d1

Reply:
Seems it's still Qt6 issue. I was just confused by the fact that it was explicitly set earlier, when we weren't looking towards Qt6 yet.

Reply:
well, they never mention that default is horizontal, so I don't see anything wrong with explicitly setting this flag for qt5, plus with different types for style available I think it's better to explicitly set this flag.

Reply:
Sounds reasonable. So I would probably refrain from adding a comment.

Reply:
>though I doubt it's just a '"workaround for Qt 6". QStyle::State_Horizontal was explicitly set for some reason before you drop/miss it out.

IIRC there was a bug in Qt5 and I (or someone else) introduced workaround and Qt fixed it afterwards, now Qt6 has the regression.

>well, they never mention that default is horizontal

If it is not horizontal then what is it? undefined, vertical? that doesn't make sense.
The state comes from `QStyleOptionViewItem`, it is expected that Qt set the appropriate flags.

BTW QProgressBar is horizontal by default: https://doc.qt.io/qt-5/qprogressbar.html#orientation-prop

>Sounds reasonable. So I would probably refrain from adding a comment.

Please consider adding one, it is unexpected to see the needing to add such flag.

Reply:
Also I won't mark this PR as a hotfix. Progress bar isn't broken on Qt5.

Reply:
>Please consider adding one, it is unexpected to see the needing to add such flag in qbt code.

`Qt 6 requires QStyle::State_Horizontal to be set for correctly drawing horizontal progress bar` could fit you?

Reply:
Sure.